test/end-to-end/core.sh

```diff
@@ -109,6 +109,22 @@ os::cmd::expect_success "curl -f http://${DOCKER_REGISTRY}/healthz"
 os::cmd::expect_success "dig @${API_HOST} docker-registry.default.local. A"
 registry_pod="$(oc get pod -n default -l deploymentconfig=docker-registry --template='{{(index .items 0).metadata.name}}')"
 
+os::log::info "Configure registry to disable mirroring"
+os::cmd::expect_success "oc project '${CLUSTER_ADMIN_CONTEXT}'"
+os::cmd::expect_success 'oc env -n default dc/docker-registry REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_MIRRORPULLTHROUGH=false'
+os::cmd::expect_success 'oc env -n default dc/docker-registry REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_PULLTHROUGH=true'
+os::cmd::expect_success 'oc rollout status dc/docker-registry'
+os::log::info "Registry configured to disble mirroring"
+
+# verify local image based on remote image can be pushed to local registry
+os::log::info "Check the image based on a non-local base image."
+os::cmd::expect_success "oc login -u user0 -p pass"
+os::cmd::expect_success "oc new-project project0"
+os::cmd::expect_success "oc new-build -D \$'FROM busybox:glibc\nRUN echo abc >/tmp/foo'"
+os::cmd::try_until_success "oc get build/busybox-1 -o go-template={{.metadata.name}} 2>&1 |grep busybox-1"
+os::cmd::try_until_not_text "oc get build/busybox-1 -o go-template='{{.status.phase}}' 2>&1" "^(Pending|Running)\$" "$(( 10*TIME_MIN ))"
+os::cmd::expect_success_and_text "oc get build/busybox-1 -o go-template='{{.status.phase}}'" '^Complete$'
```

Comment:
@stevekuznetsov 

Just after `oc new-build ...` output of `oc get builds` looks like this:
```
RUN: oc get builds
No resources found.

RUN: oc get builds
NAME        TYPE      FROM         STATUS    STARTED   DURATION
busybox-1   Docker    Dockerfile   Pending             

RUN: oc get builds
NAME        TYPE      FROM         STATUS    STARTED   DURATION
busybox-1   Docker    Dockerfile   Pending             

RUN: oc get builds
NAME        TYPE      FROM         STATUS    STARTED        DURATION
busybox-1   Docker    Dockerfile   Running   1 second ago   

RUN: oc get builds
NAME        TYPE      FROM         STATUS     STARTED         DURATION
busybox-1   Docker    Dockerfile   Complete   2 seconds ago   1s
```

If I leave only the `os::cmd::try_until_text "oc get build/busybox-1 -o jsonpath='{.status.phase}'" 'Complete' "$(( 10*TIME_MIN ))"`, in case of an error, the test will wait a very long time.

In my test first `os::cmd::try_until_success` waits for the appearance of the build. Second `os::cmd::try_until_not_text` waits for any completion of build. And the last `os::cmd::expect_success_and_text` checks status of build.

/cc @mfojtik